### PR TITLE
Support to Get IKE Security Associations on Virtual Network Gateway Connections

### PR DESCRIPTION
**Pull request by abhi7860**
## Description

Customers can now fetch IKE Security Associations for their Virtual Network Gateway Connections

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/803

    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by isra-fel — CHANGES_REQUESTED** [review]
A few things missing in your PR:

- You are using some new API `IVirtualNetworkGatewayConnectionsOperations.GetIkeSas()` that is available in the SDK on `network-august` branch right now. Please check in your local SDKs (and any related changes) to unblock build
- You need tests to cover the new command

**Comment by abhi7860** [comment]
Hi @isra-fel ,
I've added the required csproj files and packages.
Since this is only a GET operation I've added the test in another test.

File: src/Network/Network.Test/ScenarioTests/VirtualNetworkGatewayConnectionTests.ps1 [hunk 1]
```diff
@@ -774,3 +774,67 @@ function Test-VirtualNetworkGatewayConnectionPacketCapture
...
+      $localnetGateway = Get-AzLocalNetworkGateway -ResourceGroupName $rgname -name $localnetName
+      Assert-NotNull $localnetGateway
+
+      # Create & Get VirtualNetworkGatewayConnection
+      $actual = New-AzVirtualNetworkGatewayConnection -ResourceGroupName $rgname -name $vnetConnectionName -location $location -VirtualNetworkGateway1 $vnetGateway -LocalNetworkGateway2 $localnetGateway -ConnectionType IPsec -RoutingWeight 3 -SharedKey abc -EnableBgp $true
+      $connection = Get-AzVirtualNetworkGatewayConnection -ResourceGroupName $rgname -name $vnetConnectionName
+      Assert-NotNull $connection
+
+      $ikesa = Get-AzVirtualNetworkGatewayConnectionIkeSa -InputObject $connection
+      Assert-NotNull $ikesa
```
Comment (isra-fel) [inline src/Network/Network.Test/ScenarioTests/VirtualNetworkGatewayConnectionTests.ps1:828]: You may wait for a moment before getting ikesa, if it take some time for it to have value
```suggestion
      Wait-Seconds 100
      $ikesa = Get-AzVirtualNetworkGatewayConnectionIkeSa -InputObject $connection
      Assert-NotNull $ikesa
```

File: src/Network/Network.Test/ScenarioTests/VirtualNetworkGatewayConnectionTests.ps1 [hunk 2]
```diff
@@ -774,3 +774,67 @@ function Test-VirtualNetworkGatewayConnectionPacketCapture
...
+    
+      # Delete VirtualNetworkGatewayConnection
+      $delete = Remove-AzVirtualNetworkGatewayConnection -ResourceGroupName $actual.ResourceGroupName -name $vnetConnectionName -PassThru -Force
+      Assert-AreEqual true $delete
```
Comment (isra-fel) [inline src/Network/Network.Test/ScenarioTests/VirtualNetworkGatewayConnectionTests.ps1:832]:
```suggestion
      Assert-True $delete
```

**Comment by isra-fel** [comment]
Hi @abhi7860 as @MikhailTryakhov 's PR #13927 already merged `network-august` branch to `master`, `network-august` branch became stale. Any later PR (including this one) merged into `network-august` won't be included in this release.
Please edit the base branch of your PR to `release-2021-01-19` if you wish to catch this release

**Review by isra-fel — APPROVED** [review]
LGTM!

**Pull request merged**